### ISSUE TITLE
6 Engineers per Attack

### ISSUE THREAD

**Issue by Swartonf** [UnofficialCrusaderPatch/UnofficialCrusaderPatch#847]
While i was changing my AIC, i thought about the SiegeEngines.
At the moment, the AI tries to recruit 4 engineers per attack. That makes trebeuchet complicated to use. After a minimum of 4 attacks the AI uses every siegeengine, if you only use catapults or firebalista that is. If we use trebuchets or other big Siegeengines, we need 8 whole attacks to get all of our siegeengines.

To reduce the amount of attacks we need AND to give us some more options, i thought about 6 Engineers per Attack.
That would allow us to use:

- 2 Trebuchets                                                           instead of just 1 Trebeuchet per attack
- 1 Siegetower/Battering Ram+Catapult/Firebalista instead of just 1 Siegetower/Battering Ram

This would make the biggest difference for Trebeuchet AI's. With that beeing said, the following example takes 8 attacks with the  current patch. With the new option, this would take 4 attacks. In my opinion this would increase the pressure of every attack and it increases the options to strengthen the AIC.

				"SiegeEngine1": "Trebuchet",
				"SiegeEngine2": "Trebuchet",
				"SiegeEngine3": "Trebuchet",
				"SiegeEngine4": "Trebuchet",
				"SiegeEngine5": "Trebuchet",
				"SiegeEngine6": "Trebuchet",
				"SiegeEngine7": "Trebuchet",
				"SiegeEngine8": "Trebuchet",

**Comment by ByBurton** [comment]
Nah, either make it a number you can change and not just a set 6 or leave it vanilla.

**Comment by GRhin** [comment]
Totally agree that variable is best option. 

However I feel like an optional tickbox (which all ucp features are) would be better than forced vanilla. The reasoning swartonf laid out was valid and this would improve the AIs for that scenario. An optional conditional improvement is better than no improvement. 
Now this isn't to say that this feature should be a priority, outside of my wheelhouse to comment on that. Just saying that I don't see a downside to implementing this.

**Comment by ilosevsrat** [comment]
there is one way to make ai buil all 8 enjines from first attack thugh.

set attactroopbase 0.   and make attack troops random something between 30 and 100.

this way ai builds all 8 siege engines from first attack.

**Comment by OverlordMatty** [comment]
sounds good +1
